File: examples/scripting-ctx/validate.py
# Post-request hook (Python). Runs AFTER the request; `ctx.response` is the response.
# ctx.fail(msg) fails the step (even on a 200); ctx.log writes to the run output.
if ctx.response.status != 200:
    ctx.fail("expected 200, got " + str(ctx.response.status))
post = ctx.response.json
if str(post["id"]) != ctx.vars["postId"]:
    ctx.fail("id mismatch: " + str(post["id"]) + " != " + ctx.vars["postId"])
ctx.log("[validate] post " + str(post["id"]) + " ok in " + str(ctx.response.duration_ms) + "ms")

File: examples/scripting-ctx/validate.js
// Post-request hook (JavaScript). Runs AFTER the request; `ctx.response` is the response.
// ctx.fail(msg) fails the step (even on a 200); ctx.log writes to the run output.
if (ctx.response.status !== 200) {
  ctx.fail("expected 200, got " + ctx.response.status);
}
const post = ctx.response.json;
if (String(post.id) !== ctx.vars.postId) {
  ctx.fail("id mismatch: " + post.id + " != " + ctx.vars.postId);
}
ctx.log("[validate] post " + post.id + " ok in " + ctx.response.durationMs + "ms");
